charts/spire/charts/spiffe-oidc-discovery-provider/templates/configmap.yaml

```diff
@@ -1,44 +1,43 @@
 {{- $oidcSocket := "/run/spire/oidc-sockets/spire-oidc-server.sock" }}
+{{- define "spiffe-oidc-discovery-provider.yaml-config" -}}
+log_level: {{ .Values.config.logLevel | quote }}
+
+domains:
+  - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}"
+  - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ .Release.Namespace }}"
+  - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local"
+  {{- if gt (len .Values.config.domains) 0 }}
+  {{- .Values.config.domains | toYaml | nindent 2 }}
+  {{- end }}
+
+{{- if .Values.insecureScheme.enabled }}
+allow_insecure_scheme: {{ .Values.insecureScheme.enabled }}
+listen_socket_path: {{ .oidcSocket | quote }}
+{{- else }}
+acme:
+  directory_url: {{ .Values.config.acme.directoryUrl | quote }}
+  cache_dir: {{ .Values.config.acme.cacheDir | quote }}
+  tos_accepted: {{ .Values.config.acme.tosAccepted }}
+  email: {{ .Values.config.acme.emailAddress | quote }}
+{{- end }}
+
+workload_api:
+  socket_path: "/spiffe-workload-api/{{ include "spire.agent-socket-path" . | base }}"
+  trust_domain: "{{ .Values.trustDomain }}"
+
+health_checks:
+  bind_port: "8008"
+  ready_path: "/ready"
+  live_path: "/live"
+{{- end }}
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ include "spiffe-oidc-discovery-provider.fullname" . }}
   namespace: {{ .Release.Namespace }}
 data:
   oidc-discovery-provider.conf: |
-    log_level = "{{ .Values.config.logLevel }}"
-
-    domains = [
-      "{{ include "spiffe-oidc-discovery-provider.fullname" . }}",
-      "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ .Release.Namespace }}",
-      "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local",
-    {{- if gt (len .Values.config.domains) 0 }}
-      "{{- join "\",\n      \"" .Values.config.domains }}"
-    {{- end }}
-    ]
-
-    {{- if .Values.insecureScheme.enabled }}
-    allow_insecure_scheme = {{ .Values.insecureScheme.enabled }}
-    listen_socket_path = {{ $oidcSocket | quote }}
-    {{- else }}
-    acme {
-      directory_url = "{{ .Values.config.acme.directoryUrl }}"
-      cache_dir = "{{ .Values.config.acme.cacheDir }}"
-      tos_accepted = {{ .Values.config.acme.tosAccepted }}
-      email = "{{ .Values.config.acme.emailAddress }}"
-    }
-    {{- end }}
-
-    workload_api {
-      socket_path = "/spiffe-workload-api/{{ include "spire.agent-socket-path" . | splitList "/" | last }}"
-      trust_domain = "{{ .Values.trustDomain }}"
-    }
-
-    health_checks {
-      bind_port = "8008"
-      ready_path = "/ready"
-      live_path = "/live"
-    }
+    {{- include "spiffe-oidc-discovery-provider.yaml-config" (merge (dict) (dict "oidcSocket" $oidcSocket) .) | fromYaml | toPrettyJson | nindent 4 }}
```

Comment:
Instead of doing the dict/merge stuff here, you could just move the definition down one line into the macro definition

Reply:
No we can't, we need that variable as well in the nginx.conf. Thought about that as well initially.

Reply:
hmm... I see...
what about if we moved it to a macro and included it where needed? That might clean it up a bit.

Barring that, I think the (dict) piece is making a blank dictionary then merging it in, so essentially dead code?

Reply:
```suggestion
    {{- include "spiffe-oidc-discovery-provider.yaml-config" (dict "oidcSocket" $oidcSocket "root" .) | fromYaml | toPrettyJson | nindent 4 }}
```

Reply:
That dict is to ensure the right type. Adding another level of indirection does not make it more clear to me. struggling to understand the suggestion.